SmallStatusWidget/SmallStatusWidget.swift

```diff
@@ -0,0 +1,239 @@
+//
+//  SmallStatusWidget.swift
+//  Loop
+//
+//  Created by Noah Brauner on 8/15/22.
+//  Copyright © 2022 LoopKit Authors. All rights reserved.
+//
+
+import WidgetKit
+import SwiftUI
+import LoopKit
+import LoopCore
+import HealthKit
+import CoreData
+
+class Provider: TimelineProvider {
+    lazy var defaults = UserDefaults.appGroup
+    
+    lazy var healthStore = HKHealthStore()
+
+    lazy var cacheStore = PersistenceController.controllerInAppGroupDirectory()
+
+    lazy var localCacheDuration = Bundle.main.localCacheDuration
+
+    lazy var settingsStore: SettingsStore =  SettingsStore(
+        store: cacheStore,
+        expireAfter: localCacheDuration)
+
+    lazy var glucoseStore = GlucoseStore(
+        healthStore: healthStore,
+        observeHealthKitSamplesFromOtherApps: FeatureFlags.observeHealthKitGlucoseSamplesFromOtherApps,
+        storeSamplesToHealthKit: false,
+        cacheStore: cacheStore,
+        observationEnabled: false,
+        provenanceIdentifier: HKSource.default().bundleIdentifier
+    )
+
+    func placeholder(in context: Context) -> SmallStatusEntry {
+        return SmallStatusEntry(date: Date(), lastLoopCompleted: nil, closeLoop: true, currentGlucose: nil, previousGlucose: nil, unit: .milligramsPerDeciliter, sensor: nil, netBasal: nil, eventualGlucose: nil, minsAgo: 5)
+    }
+
+    func getSnapshot(in context: Context, completion: @escaping (SmallStatusEntry) -> ()) {
+        update { newEntry in
+            completion(newEntry)
+        }
+    }
+    
+    func getTimeline(in context: Context, completion: @escaping (Timeline<Entry>) -> ()) {
+        update { newEntry in
+            var entries = [newEntry]
+            
+            let midnight = Calendar.current.startOfDay(for: Date())
+            let nextHour = midnight.addingTimeInterval(.hours(1))
+            
+            for minute in 1..<60 {
```

Comment:
Done, see above. Rather than marking the loop circle as yellow or red when it could potentially be green, the whole widget is grayed if it doesn't refresh on a loop. The only rendered frames are the initial one, the stale glucose, and the stale widget.